transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -583,16 +586,21 @@ private void callHandlerAdded0(final AbstractChannelHandlerContext ctx) {
     }
 
     private void callHandlerRemoved(final AbstractChannelHandlerContext ctx) {
-        if (isExecuteLater(ctx)) {
-            ctx.executor().execute(new OneTimeTask() {
-                @Override
-                public void run() {
-                    callHandlerRemoved0(ctx);
-                 }
-            });
-            return;
+        EventExecutor executor = ctx.executorSafe();
+        if (executor != null) {
+            if (executor.inEventLoop()) {
+                callHandlerRemoved0(ctx);
+            } else {
+                ctx.executor().execute(new OneTimeTask() {
```

Comment:
@Scottmitch ah ok.. No it should be changed to jus use `executor ` (as you suggested in your first comment).